### ISSUE TITLE
S3 Source connector does not preserve partitions

### ISSUE THREAD

**Issue by YordanPavlov** [lensesio/stream-reactor#868]
### What version of the Stream Reactor are you reporting this issue for?

Master build

### What is the expected behaviour?

I have the following original Kafka topic I want to backup in S3:

```
test_yordan_kafka_connect:0:24004
test_yordan_kafka_connect:1:48830
test_yordan_kafka_connect:2:27166
```

Once the backup is completed I can see the files in s3 as expected:

```
test-bucket/test_yordan_kafka_connect/0
[2022-07-06 11:38:35 UTC] 1.8MiB STANDARD 14999.json
[2022-07-06 11:38:36 UTC] 1.8MiB STANDARD 19999.json
[2022-07-06 11:38:31 UTC] 1.8MiB STANDARD 4999.json
[2022-07-06 11:38:33 UTC] 1.8MiB STANDARD 9999.json

test-bucket/test_yordan_kafka_connect/1
[2022-07-06 11:38:34 UTC] 1.8MiB STANDARD 14999.json
[2022-07-06 11:38:36 UTC] 1.8MiB STANDARD 19999.json
[2022-07-06 11:38:37 UTC] 1.8MiB STANDARD 24999.json
[2022-07-06 11:38:38 UTC] 1.8MiB STANDARD 29999.json
[2022-07-06 11:38:38 UTC] 1.8MiB STANDARD 34999.json
[2022-07-06 11:38:39 UTC] 1.8MiB STANDARD 39999.json
[2022-07-06 11:38:39 UTC] 1.8MiB STANDARD 44999.json
[2022-07-06 11:38:30 UTC] 1.8MiB STANDARD 4999.json
[2022-07-06 11:38:32 UTC] 1.8MiB STANDARD 9999.json

...
```

### What was observed?

Once the data is restored in a new Kafka topic using the Source connector I am seeing the following offsets:
```
test_yordan_kafka_connect_copy:0:30347
test_yordan_kafka_connect_copy:1:30058
test_yordan_kafka_connect_copy:2:29595
```
notice how the offsets are evenly distributed unlike the original topic.

### What is your connector properties configuration (my-connector.properties)?
```json
{
  "connector.class":"io.lenses.streamreactor.connect.aws.s3.source.S3SourceConnector",
  "tasks.max":"1",
  "aws.auth.mode": "Default",
  "connect.s3.kcql": "insert into test_yordan_kafka_connect_copy select * from test-flink-savepoints-test-hz-stage:test-bucket/test_yordan_kafka_connect STOREAS `json`",
  "connect.s3.aws.client": "AWS",
  "connect.s3.aws.region": "eu-central-1"
}
```

### Please provide full log files (redact and sensitive information)
[source-log.txt](https://github.com/lensesio/stream-reactor/files/9054667/source-log.txt)

P.S. I have some more questions if you allow
1. The documentation says:
"Please note: The JSON is not parsed by the S3 Source connector. There is no difference in handling between Json and Text by the S3 Source connector." Is there a way to make the Kafka record be stored as JSON instead of Text?
2. I have schema-less JSON data, It seems I can't use Avro and Parquet, what would be a way to compress the data in S3, can I enable gzip compression on the sink?

**Comment by iosifnicolae2** [comment]
Another solution would be to use https://github.com/lensesio/stream-reactor/issues/869#issuecomment-1234586024
How it works:
 - you apply the `com.github.jcustenborder.kafka.connect.archive.Archive` transformer on the Sink Connector (the key, value and partition data in S3: https://github.com/iosifnicolae2/kafka-connect-transform-archive/blob/10aa03edef4919c78c631bac37bf2746a5a1a9d5/src/main/java/com/github/jcustenborder/kafka/connect/archive/Archive.java#L72 )
 - you apply `com.github.jcustenborder.kafka.connect.archive.UnArchive` transformer on the Source Connector (everything will be restored correctly: https://github.com/iosifnicolae2/kafka-connect-transform-archive/blob/10aa03edef4919c78c631bac37bf2746a5a1a9d5/src/main/java/com/github/jcustenborder/kafka/connect/archive/UnArchive.java#L39)

Note. The target topic must have the correct number of partitions before starting the Source Connector.

**Comment by davidsloan** [comment]
PR to fix the config issue: https://github.com/lensesio/stream-reactor/pull/900

**Comment by davidsloan** [comment]
The fix has been released in 4.1.0
